src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Visitors/ListObjectTypeVisitor.cs

```diff
@@ -54,14 +56,21 @@ public override void Visit(IAcceptor acceptor, KeyValuePair<string, Type> type,
             };
             var schemas = new Dictionary<string, OpenApiSchema>();
 
-            var subAcceptor = new OpenApiSchemaAcceptor()
+            OpenApiSchemaAcceptor subAcceptor;
+            if (!this.visitedTypes.ContainsKey(underlyingType))
             {
-                Types = types,
-                RootSchemas = instance.RootSchemas,
-                Schemas = schemas,
-            };
+                subAcceptor = new OpenApiSchemaAcceptor()
+                {
+                    Types = types, RootSchemas = instance.RootSchemas, Schemas = schemas,
+                };
+                this.visitedTypes.Add(underlyingType, subAcceptor);
+                subAcceptor.Accept(this.VisitorCollection, namingStrategy);
 
-            subAcceptor.Accept(this.VisitorCollection, namingStrategy);
+            }
+            else
+            {
+                subAcceptor = this.visitedTypes[underlyingType];
+            }
```

Comment:
I'd like to ask you to add this logic to the `DictionaryObjectTypevisitor` as well.